blogposts/2021/why-code-reviews-are-hard.md

```diff
@@ -0,0 +1,98 @@
+
+Another advantage is that junior developers can ask questions and get answers from experienced developers without much friction. This way, the formal code review process also acts as a case study in learning how to code.
+
+With a larger group performing the review, the technical focus should also shift. Usually, groups are more into the “why” than the “how.” This leads to investigating if the desired behavior is correctly implemented, and if edge cases are handled. Sometimes, it also identifies potential missing cases and scenarios to consider in future implementation. The review then produces a direct positive business impact.
+
+To reduce larger code reviews, timebox each review session. Always determine its focus in the beginning, and check each file. So, the change creator sees completed reviews and what is still left after each review session.
+
+But, what about context? How can we improve context retrieval? An enhanced code search is helpful here.
+
+### Add context to the review process 
+
+Universal code search tools help to ease code review pain by providing reviewers with the right context, whether implicitly or on demand. Finding all references or jumping to definitions is only the beginning. There’s also no need to spin up a local integrated development environment.
+
+Universal code search combines different repositories using different project types and languages in one search engine. The result is a knowledge graph of all combined sources that you can query from a single location. This is done using a set of smart filters and context-aware operators. The search engine understands relationships between different parts of the code and treats them correctly.
+
+A solution like Sourcegraph allows you to enhance existing tools such as GitHub or GitLab with universal search integration. Integrations are a vital part of the expected developer experience. Without the necessary integrations, developers may need to switch tooling.
+
+With universal code search running on GitHub, more context is available directly in the code review. Method or function names link to corresponding sections of the codebase. There’s no need to keep a separate copy of the code open in your editor or integrated development environment to follow along and explore the changes.
+
+Search features in integrated development environments and text editors aren’t always up to the task of finding what’s needed. For example, code might be present many times, in different locations, in a database. Universal code search’s ability to spot similar code or duplicates is useful: it identifies existing code much better and infers potential refactoring paths.
+
+These universal code search abilities also reduce the impact of large code reviews. Reviewers can see and understand larger change sets.
```

Comment:
@content-hacker I'm concerned about this section going a bit too hard on pushing Sourcegraph and putting people off. I'd like to cut it down a bit and see if we can introduce 1 or 2 other suggestions for improving context in code reviews. WDYT?

Reply:
@content-hacker After our 1:1 today I'm thinking we should hold off on this post to see if it aligns with our current thinking/how we might be able to improve on it.